tests/online_migration/sle12_online_migration/zypper_patch.pm

```diff
@@ -0,0 +1,36 @@
+# SLE12 online migration tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+    my $self = shift;
+    select_console 'root-console';
+
+    # fully update the system
+    script_run("zypper -n patch --with-interactive -l; echo 'worked-patch-\$?' > /dev/$serialdev", 0);
+
+    my $ret = wait_serial "worked-patch-\?-", 700;
+    $ret =~ /worked-patch-(\d+)/;
+    die "zypper failed with code $1" unless $1 == 0 || $1 == 102 || $1 == 103;
+
+    script_run("zypper -n patch --with-interactive -l; echo 'worked-2-patch-\$?-' > /dev/$serialdev", 0);    # first one might only have installed "update-test-affects-package-manager"
```

Comment:
as you copy & pasted the same broken code (\? in '' will not give the exit status) - please port to the new zypper_call function in utils